sdk/python/tests/compiler/compiler_tests.py

```diff
@@ -711,6 +716,29 @@ def some_pipeline():
       container = template.get('container', None)
       if container:
         self.assertEqual(template['retryStrategy']['limit'], 5)
+        
+  def test_oob_component_label(self):
+    gcs_download_op = components.load_component_from_url(
+        _TEST_GCS_DOWNLOAD_COMPONENT_URL)
+    
+    @dsl.pipeline(name='some_pipeline')
+    def some_pipeline():
+      _download_task = gcs_download_op('gs://some_bucket/some_dir/some_file')
+      
+    workflow_dict = compiler.Compiler()._compile(some_pipeline)
+    
+    found_download_task = False
+    for template in workflow_dict['spec']['templates']:
+      if template['name'] == 'download-from-gcs':
```

Comment:
It might be less flaky to not depend on the template name. You can use `if getattr(template, 'container', None) `

Reply:
Done